addon/poll-task.js

```diff
@@ -158,15 +158,32 @@ export function pollTask(obj, taskOrName, token = getNextToken()) {
    @param { String } token the token for the pollTask to be cleared
    @public
    */
-export function cancelPoll(token) {
+export function cancelPoll(obj, token) {
+  // TODO: remove this older API on next major version
+  if (token === undefined) {
+    deprecate(
+      'ember-lifeline cancelPoll called without an object. New syntax is cancelPoll(obj, cancelId) and avoids a memory leak.',
+      true,
+      {
+        id: 'ember-lifeline-cancel-poll-without-object',
+        until: '4.0.0',
+      }
+    );
+    token = obj;
+    obj = null;
+  }
   delete queuedPollTasks[token];
+  if (obj) {
+    let pollers = registeredPollers.get(obj);
+    pollers.delete(token);
+  }
 }
 
 function getPollersDisposable(pollers) {
   return function() {
-    for (let i = 0; i < pollers.length; i++) {
-      delete queuedPollTasks[pollers[i]];
-    }
+    pollers.forEach(token => {
+      cancelPoll(token);
```

Comment:
Doesn't this trigger the deprecation during destruction of every object that uses polling? In order to add the deprecation, we have to ensure that our own internal usages do _not_ trigger it...

@scalvert - We should bring in some additional test helper utilities to ensure none of our tests trigger deprecations ([example here](https://github.com/emberjs/ember-test-helpers/blob/7f9a3d082958ba993c6db863886d370d73fd98ef/tests/test-helper.js#L36-L102))...

Reply:
Agree.